src/main/java/wellnus/reflection/feature/QuestionList.java

```diff
@@ -148,18 +148,25 @@ public void storeFavList() throws StorageException {
      * @throws StorageException If there is error during tokenization
      * @throws TokenizerException If there is error during detokenization
      */
-    public void loadFavList() throws StorageException, TokenizerException {
-        ArrayList<String> loadedFavList = storage.loadData(FILE_NAME);
-        ArrayList<Set<Integer>> detokenizedFavList = reflectionTokenizer.detokenize(loadedFavList);
-        this.setDataIndex(detokenizedFavList);
+    public void loadQuestionData() throws StorageException, TokenizerException {
+        ArrayList<String> loadedQuestionList = storage.loadData(FILE_NAME);
+        ArrayList<Set<Integer>> detokenizedQuestionList = reflectionTokenizer.detokenize(loadedQuestionList);
+        this.setDataIndex(detokenizedQuestionList);
+        this.randomQuestionIndexes = this.dataIndex.get(INDEX_ONE);
     }
 
     /**
      * Generate a set of 5 distinct random numbers from 0-9 which will then be used as indexes to
      * select 5 random questions.
      */
-    public void setRandomQuestionIndexes() {
+    public void setRandomQuestionIndexes() throws StorageException {
         this.randomQuestionIndexes = RANDOM_NUMBER_GENERATOR.generateRandomNumbers();
+        ArrayList<Set<Integer>> updatedQuestionData = new ArrayList<>();
+        Set<Integer> favIndexList = this.dataIndex.get(INDEX_ZERO);
+        updatedQuestionData.add(favIndexList);
+        updatedQuestionData.add(this.randomQuestionIndexes);
+        this.setDataIndex(updatedQuestionData);
```

Comment:
Would it be simpler to simply do `getDataIndex().set(INDEX_ZERO, randomQuestionIndexes)`?